models/demos/falcon7b/tt/model_utils.py

```diff
@@ -57,3 +59,88 @@ def get_weights_cached(
             weights[0].cpu(),
         )
     return weights
+
+
+# this function takes activations that are assumed to be non-padded and weights that are assumed to be padded
+# (since they are pushed to device at the moment of model initialization);
+# it also takes in number of slices since this should be determined at the moment of pushing weights,
+# but in general with 512 < seq_len <= 1024 we should use 4 slices, with 1024 < seq_len <= 2048 we should use 8 slices
+def falcon_lm_head_matmul_2d(
+    hidden_states: tt_lib.tensor.Tensor,
+    weights: List[tt_lib.tensor.Tensor],
+    num_slices: int,
+    in0_mem_config: tt_lib.tensor.MemoryConfig,
+    in0_dtype: tt_lib.tensor.DataType,
+    out_mem_config: tt_lib.tensor.MemoryConfig,
+    out_dtype: tt_lib.tensor.DataType,
+):
+    seq_len = hidden_states.get_legacy_shape()[-2]
+
+    assert seq_len % 32 == 0, f"Sequence length must be a multiple of 32, instead it is {seq_len}"
+    assert seq_len > 512, f"Falcon lm head 2d is only supported for sequence length > 512, instead it is {seq_len}"
+    assert seq_len <= 2048, f"Falcon lm head 2d is only supported for sequence length <= 2048, instead it is {seq_len}"
+
+    assert (
+        len(weights) == num_slices
+    ), f"Weights are expected to be split into {num_slices} slices, instead there are {len(weights)}"
+    weights_inner_dim_in_tiles = weights[0].get_legacy_shape()[-2] // 32
+    assert (
+        weights_inner_dim_in_tiles == 144
+    ), f"Weights are expected to be padded to the inner dim 144 in tiles, instead they are {weights_inner_dim_in_tiles}"
+
+    # pad activations to inner dim 144
+    padding = torch.zeros([1, 1, seq_len, 64])
+    padding_t = (
+        tt_lib.tensor.Tensor(padding, in0_dtype)
+        .to(tt_lib.tensor.Layout.TILE)
+        .to(hidden_states.device(), in0_mem_config)
+    )
+    hidden_states = tt_lib.tensor.concat([hidden_states, padding_t], -1)
+
+    compute_kernel_config = tt_lib.tensor.WormholeComputeKernelConfig(
```

Comment:
Since this function supports only wormhole, it would be good to prevent usage of it by Grayskull demo/tests